### PR TITLE
Add tests for `REPLMode.promptf`

### PR DESCRIPTION
**Pull request by 00vareladavid**
Clean version of #466 . Note it should also fix the Apple bug.

### CODE REVIEW

File: test/utils.jl [hunk 1]
```diff
@@ -54,3 +54,15 @@ function with_current_env(f)
         Pkg.activate()
     end
 end
+
+function with_temp_env(f, env_name::AbstractString="Dummy")
+    TEST_SIG = LibGit2.Signature("TEST", "TEST@TEST.COM", round(time()), 0)
```
[the remark this answers is not in the document]
Comment (00vareladavid) [reply test/utils.jl]: Yeah I forgot to remove that :grin:

File: test/repl.jl [hunk 2]
```diff
@@ -437,4 +437,43 @@ end
...
+        @test Pkg.REPLMode.promptf() == "($env_name) pkg> "
+
+        newname = "NewNameII"
+        set_name(projfile_path, newname)
+        cd(env_path) do
+            @test Pkg.REPLMode.promptf() == "($newname) pkg> "
+        end
+        @test Pkg.REPLMode.promptf() == "($newname) pkg> "
+
+        #rm(projfile_path)
```
Comment (KristofferC) [inline test/repl.jl]: Should this be removed?

**Comment by 00vareladavid** [comment]
That unlink error has been resurfacing a lot recently :/ , but this PR should be good for a merge if everything else is OK.

**Comment by KristofferC** [comment]
Yes, I will have to try to debug it today... Not fun.

**Comment by KristofferC** [comment]
Sorry, I introduced some merge conflict here :/

**Comment by KristofferC** [comment]
Ok, should be fixed.

**Comment by KristofferC** [comment]
Thanks!

**Comment by 00vareladavid** [comment]
Welcome!

**Pull request merged**